src/action_state.rs

```diff
@@ -125,70 +169,149 @@ impl ActionData {
 /// ```
 #[derive(Resource, Component, Clone, Debug, PartialEq, Serialize, Deserialize, Reflect)]
 pub struct ActionState<A: Actionlike> {
-    /// The [`ActionData`] of each action
-    action_data: HashMap<A, ActionData>,
+    /// The [`ButtonData`] of each action
+    button_data: HashMap<A, ButtonData>,
+    /// The [`AxisData`] of each action
+    axis_data: HashMap<A, AxisData>,
```

Comment:
Is it intended that an action could be associated with both an AxisData and a ButtonData?

Reply:
Oh, Button-presses, 1-d joysticks, and 2-d joysticks are now completely independent?

Reply:
Yes, they are. Generally you won't reasonably end up with an action associated with more than one of the input types though.

Reply:
I wonder if some metadata should be shared between all 3 options.
For example the `ActionState` being disabled seems more-or-less independent from the way the action is mapped (button/axis/axis-2d).

For example in my case I want to stop input replication systems if the Action is disabled. Now it looks like i would have to check independently for all 3 subtypes of the action

Reply:
Right 🤔 I didn't think about that super hard, but I suspect that we should be able to refactor for that... I think probably in a follow-up: I want to get this PR out the door before it grows any larger.